python/models/hierarchical_filtering_proxy_model.py

```diff
@@ -471,8 +532,41 @@ def _on_source_model_rows_inserted(self, parent_idx, start, end):
             # invalid input parameters so ignore!
             return
 
-        # dirty specific rows in the caches:
-        self._dirty_accepted_rows(parent_idx, start, end)
+        # Get the current value for the parent
+        cached_value = self._rejected_cache.get(parent_idx)
+        # Clear the cache value for the parent.
+        self._rejected_cache.remove(parent_idx)
```

Comment:
what about siblings? are they removed too? Maybe worth clarifying.

Reply:
will add a comment, siblings are not affected.